File: iotcloud-restapi-php/Core/src/IoTCloud/Response/ListDevicesResponse.php
<?php

namespace TXIoTCloud\Response;

require_once "BaseResponse.php";

class ListDevicesResponse extends BaseResponse
{
    /**
     * 设备总数
     */
    private $totalCnt;

    /**
     * 设备对象的数组，数组元素为DeviceInfo实例
     */
    private $devices;

    public function __construct()
    {
    }

    /**
     * 获取设备总数
     */
    public function getTotalCnt()
    {
        return $this->totalCnt;
    }

    /**
     * 设置设备总数
     * @param $totalCnt
     */
    public function setTotalCnt($totalCnt)
    {
        $this->totalCnt = $totalCnt;
    }

    /**
     * 获取设备对象的数组
     * @return array 数组元素为DeviceInfo实例
     */
    public function getDevices()
    {
        return $this->devices;
    }

    /**
     * 设置设备对象的数组
     * @param $devices
     */
    public function setDevices($devices)
    {
        $this->devices = $devices;
    }


}

File: iotcloud-restapi-java/Core/src/com/qcloud/restapi/iotcore/response/ListDevicesResponse.java
package com.qcloud.restapi.iotcore.response;

import com.google.gson.*;

import java.util.ArrayList;
import java.util.List;

public class ListDevicesResponse extends Response {
    public class DeviceInfo {
        public String deviceName= "";

        public DeviceInfo() {

        }
    }

    /**
     * 解析JSON字串并构造ListDevicesResponse对象
     *
     * @param response 从服务端返回的应答JSON字串
     */
    public ListDevicesResponse(String response) {
        super(response);
    }

    public List <DeviceInfo> getDevices() {
        List <DeviceInfo> list = new ArrayList<>();

        try {
            JsonArray arr = jsonObj.getAsJsonArray("devices");
            for (int i = 0; i < arr.size(); i++) {
                DeviceInfo info = new DeviceInfo();

                info.deviceName = arr.get(i).getAsJsonObject().get("deviceName").getAsString();

                list.add(info);
            }
        }catch (Exception e) {
        }

        return list;
    }
}
